stores/transaction/src/main/java/com/bloxbean/cardano/yaci/store/transaction/scheduler/TransactionPruningService.java

```diff
@@ -0,0 +1,84 @@
+package com.bloxbean.cardano.yaci.store.transaction.scheduler;
+
+
+import com.bloxbean.cardano.yaci.store.common.service.CursorService;
+import com.bloxbean.cardano.yaci.store.events.EpochChangeEvent;
+import com.bloxbean.cardano.yaci.store.transaction.TransactionStoreProperties;
+import com.bloxbean.cardano.yaci.store.transaction.storage.TransactionStorage;
+import com.bloxbean.cardano.yaci.store.transaction.storage.TransactionWitnessStorage;
+import jakarta.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Component
+@ConditionalOnProperty(
+        value = "store.transaction.pruning-enabled",
+        havingValue = "true"
+)
+@RequiredArgsConstructor
+@Slf4j
+public class TransactionPruningService {
+    private final TransactionStorage transactionStorage;
+    private final TransactionWitnessStorage transactionWitnessStorage;
+    private final CursorService cursorService;
+    private final TransactionStoreProperties transactionStoreProperties;
+
+    private final AtomicBoolean isPruning = new AtomicBoolean(false);
+
+    @PostConstruct
+    public void init() {
+        log.info("<< Transaction Pruning Service Enabled >>");
+    }
+
+    @Scheduled(fixedRateString = "${store.transaction.pruning-interval:86400}", timeUnit = TimeUnit.MINUTES)
```

Comment:
My suggestion is to keep the default or around 40 hours.

Reply:
So you mean we should set default is 40 hours, right?

Reply:
@matiwinnetou The current default 2160 (43200 slots) blocks is taken from the protocol parameter "securityParam" as no rollback possible after this duration. We are using the same value for other stores like utxo.

But I think, the application will set their own safe  slots based on application requirement.